src/metadatamanipulators/AddAllContentdmData.php

```diff
@@ -0,0 +1,211 @@
+          if (strlen($compound_object_info) && preg_match('/<cpd>/', $compound_object_info)) {
+              $cdata = $dom->createCDATASection($compound_object_info);
+              $dmGetCompoundObjectInfo->appendChild($cdata);
+              $contentdmdata->appendChild($dmGetCompoundObjectInfo);
+          }
+
+          // Add the <GetParent> element.
+          $GetParent = $dom->createElement('GetParent');
+          $now = $dom->createAttribute('timestamp');
+          $now->value = $timestamp;
+          $GetParent->appendChild($now);
+          $mimetype = $dom->createAttribute('mimetype');
+          $mimetype->value = 'text/xml';
+          $GetParent->appendChild($mimetype);
+          $source = $dom->createAttribute('source');
+          $source_url = $this->settings['METADATA_PARSER']['ws_url'] .
+              'GetParent/' . $this->alias . '/' . $this->record_key . '/xml';
+          $source->value = $source_url;
+          $GetParent->appendChild($source);                  
+          $parent_info = $this->getCdmData($this->alias, $this->record_key,
+              'GetParent', 'xml');
+          // Only add the <GetParent> element if the object has a parent
+          // pointer of not -1.
+          if (strlen($parent_info) && !preg_match('/\-1/', $parent_info)) {
+              $cdata = $dom->createCDATASection($parent_info);
+              $GetParent->appendChild($cdata);
+              $contentdmdata->appendChild($GetParent);
+          }             
+
+          return $dom->saveXML($dom->documentElement);
+        }
+        else {
+            // If current fragment is not <extension><CONTENTdmData>, return it
+            // unmodified.
+            return $input;
+        }
+    }
+
+    /**
+     * Fetch the output of the CONTENTdm web API for the current object.
+     *
+     * @param string $alias
+     *   The CONTENTdm alias for the current object.
+     * @param string $pointer
+     *   The CONTENTdm pointer for the current object.
+     * @param string $cdm_api_function
+     *   The name of the CONTENTdm API function.
+     * @param string $format
+     *   Either 'json' or 'xml'.
+     *
+     * @return stting
+     *   The output of the CONTENTdm API request, in the format specified.
+     */
+    private function getCdmData($alias, $pointer, $cdm_api_function, $format)
+    {
+          // Use Guzzle to fetch the output of the call to dmGetItemInfo
+          // for the current object.
+          $url = $this->settings['METADATA_PARSER']['ws_url'] .
+              $cdm_api_function . '/' . $this->alias . '/' . $pointer . '/' . $format;
+          $client = new Client();
+          try {
+              $response = $client->get($url);
+          } catch (Exception $e) {
+              $this->log->addInfo("AddContentdmData",
+                  array('HTTP request error' => $e->getMessage()));
+              return '';
+          }
+          $output = $response->getBody();
+          return $output;
+    }
+
+}
```

Comment:
I haven't done a diff but in a nutshell how is this manipulator different from https://github.com/MarcusBarnes/mik/blob/master/src/metadatamanipulators/AddContentdmData.php?